packages/xo-server/src/api/proxy.js

```diff
@@ -96,15 +96,35 @@ update.params = {
   },
 }
 
-export function deploy({ sr }) {
-  return this.deployProxy(sr._xapiId)
+export function deploy({ networkConfiguration, sr }) {
+  return this.deployProxy(sr._xapiId, { networkConfiguration })
 }
 
 deploy.permission = 'admin'
 deploy.params = {
   sr: {
     type: 'string',
   },
+  networkConfiguration: {
+    type: 'object',
+    optional: true,
+    properties: {
+      ip: {
+        type: 'string',
+      },
+      netmask: {
+        type: 'string',
+        optional: true,
+      },
+      gateway: {
+        type: 'string',
+      },
+      dns: {
+        type: 'string',
+        optional: true,
+      },
+    },
```

Comment:
Why are you a `networkConfiguration` object instead of putting all settings at the same level?

Reply:
To handle it by context, the `Ip` and the `gateway` is only required when the user want to set the network configuration.